File: 3427-sum-of-variable-length-subarrays/3427-sum-of-variable-length-subarrays.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums) {
        
        int n=nums.size();
        int sum=0;

        for(int i=0;i<n;i++){
            int start = max(0,i-nums[i]);
            for(int j=start;j<=i;j++){
                sum += nums[j];
            }
        }

        return sum;

    }
};

File: 3427-sum-of-variable-length-subarrays/3427-sum-of-variable-length-subarrays.java
class Solution {
    public int subarraySum(int[] nums) {
        int n=nums.length;
        int sum=0;
        for(int i=0;i<n;i++){
            int st=Math.max(0,i-nums[i]);
            for(int j=st;j<=i;j++){
                sum+=nums[j];

            }
        }
        return sum;
    }
}